src/data/fin/tuple/sort.lean

```diff
@@ -91,3 +90,75 @@ begin
 end
 
 end tuple
+
+namespace tuple
+
+open list
+
+variables {n : ℕ} {α : Type*}
+
+/-- If two permutations of a tuple `f` are both monotone, then they are equal. -/
+lemma unique_monotone [partial_order α] {f : fin n → α} {σ τ : equiv.perm (fin n)}
+  (hfσ : monotone (f ∘ σ)) (hfτ : monotone (f ∘ τ)) : f ∘ σ = f ∘ τ :=
+of_fn_injective $ eq_of_perm_of_sorted
+  ((σ.of_fn_comp_perm f).trans (τ.of_fn_comp_perm f).symm) hfσ.of_fn_sorted hfτ.of_fn_sorted
+
+variables [linear_order α] {f : fin n → α} {σ : equiv.perm (fin n)}
+
+lemma graph_equiv₂_apply (i : fin n) : graph_equiv₂ f i =
+  ⟨to_lex (f (sort f i), sort f i), finset.mem_image_of_mem _ $ finset.mem_univ _⟩ :=
```

Comment:
maybe this should simply be stated as:
```suggestion
lemma graph_equiv₂_apply (i : fin n) : graph_equiv₂ f i = graph_equiv₁ f (sort f i) :=
```
and moved up just below the definition of `sort`? This way it's more obvious.

Reply:
I agree.